### PR TITLE
Fix DataFormatter::formatCallerInfo to handle Closure

### PR DESCRIPTION
**Pull request by Fedik**
Pull Request for Issue #38556 .

### Summary of Changes
Correct DataFormatter::formatCallerInfo to handle Closure variable

### Testing Instructions
Please follow #38556

### Actual result BEFORE applying this Pull Request
An error

### Expected result AFTER applying this Pull Request
No error

### Documentation Changes Required
nope

### CODE REVIEW

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 04c31778a2772f0a69bfda11d9eb32e4c01d2f83<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38581">issues.joomla.org/tracker/joomla-cms/38581</a>.</sub>

File: plugins/system/debug/src/DataFormatter.php [hunk 1]
```diff
@@ -69,7 +69,16 @@ public function formatCallerInfo(array $call): string
 
             $string = rtrim($string, ', ') . ')';
         } elseif (isset($call['args'][0])) {
-            $string .= htmlspecialchars($call['function']) . ' ' . $call['args'][0];
+            $string .= htmlspecialchars($call['function']) . '(';
+
+            if (is_scalar($call['args'][0])) {
+                $string .= $call['args'][0];
+            } elseif (\is_object($call['args'][0])) {
+                $string .=  \get_class($call['args'][0]);
```
Comment (zero-24) [inline plugins/system/debug/src/DataFormatter.php]:
```suggestion
                $string .= \get_class($call['args'][0]);
```

File: plugins/system/debug/src/DataFormatter.php [hunk 2]
```diff
@@ -69,7 +69,16 @@ public function formatCallerInfo(array $call): string
...
+            } else {
+                $string .=  gettype($call['args'][0]);
```
Comment (zero-24) [inline plugins/system/debug/src/DataFormatter.php]:
```suggestion
                $string .= gettype($call['args'][0]);
```

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38581">issues.joomla.org/tracker/joomla-cms/38581</a>.</sub>

**Pull request merged**